dokka-runners/dokka-gradle-plugin/src/testFunctional/kotlin/DokkaPluginFunctionalTest.kt

```diff
@@ -144,22 +144,24 @@ class DokkaPluginFunctionalTest : FunSpec({
                 configurationsDump
                     .filter { (k, v) -> "$k=$v".contains("dokka", ignoreCase = true) }
                     .asClue { dokkaConfigurations ->
-                        expectedFormats.forEach { expectedFormat ->
 
+                        dokkaConfigurations.keys.shouldContainExactlyInAnyOrder(
+                            expectedFormats.flatMap { format ->
+                                listOf(
+                                    "Configuration dokka${format}GeneratorRuntimeResolver~internal",
+                                    "Configuration dokka${format}ModuleOutputDirectoriesResolver~internal",
+                                    "Configuration dokka${format}PluginIntransitiveResolver~internal",
+                                    "Configuration dokka${format}PublicationPluginResolver~internal",
+                                )
+                            }
+                        )
+
+                        expectedFormats.forEach { expectedFormat ->
                             val format = expectedFormat.lowercase()
 
                             @Suppress("LocalVariableName")
                             val Format = format.uppercaseFirstChar()
 
-                            dokkaConfigurations.keys.shouldContainExactlyInAnyOrder(
-                                listOf(
-                                    "Configuration dokka${Format}GeneratorRuntimeResolver~internal",
-                                    "Configuration dokka${Format}ModuleOutputDirectoriesResolver~internal",
-                                    "Configuration dokka${Format}PluginIntransitiveResolver~internal",
-                                    "Configuration dokka${Format}PublicationPluginResolver~internal",
-                                )
-                            )
-
```

Comment:
I have no idea why this test started failing, or more accurately, I have no idea how this test ever worked.

The changes in this PR should be unrelated, so they shouldn't have affected this test. But also, I don't see how this test ever worked... The Configurations for both HTML and Javadoc should always be present.

Reply:
Can we merge it via separate PR?

Reply:
https://github.com/Kotlin/dokka/pull/3801